contracts/libs/0.8.x/BytecodeStorage.sol

```diff
@@ -0,0 +1,109 @@
+// Created By: Art Blocks Inc.
+
+pragma solidity ^0.8.0;
+
+/**
+ * @title Art Blocks Script Storage Library
+ * @author Art Blocks Inc.
+ * @author Modified from 0xSequence (https://github.com/0xsequence/sstore2/blob/master/contracts/SSTORE2.sol)
+ * @author Modified from Solmate (https://github.com/transmissions11/solmate/blob/main/src/utils/SSTORE2.sol)
+ * @notice Utilize contract bytecode as persistant storage for large chunks of script string data.
+ */
+library BytecodeStorage {
+    /// always set first byte to 0x00 (STOP) to ensure created contracts cannot be called
+    uint256 internal constant DATA_OFFSET = 1;
+
+    /**
+     * @notice Write a string to contract bytecode
+     * @param _data string to be written to contract
+     * @return address_ address of deployed contract with bytecode containing concat(0x00, data)
+     */
+    function writeToBytecode(string memory _data)
+        internal
+        returns (address address_)
+    {
+        // prefix bytecode with
+        bytes memory creationCode = abi.encodePacked(
+            //---------------------------------------------------------------------------------------------------------------//
+            // Opcode  | Opcode + Arguments  | Description  | Stack View                                                     //
+            //---------------------------------------------------------------------------------------------------------------//
+            // 0x60    |  0x600B             | PUSH1 11     | codeOffset                                                     //
+            // 0x59    |  0x59               | MSIZE        | 0 codeOffset                                                   //
+            // 0x81    |  0x81               | DUP2         | codeOffset 0 codeOffset                                        //
+            // 0x38    |  0x38               | CODESIZE     | codeSize codeOffset 0 codeOffset                               //
+            // 0x03    |  0x03               | SUB          | (codeSize - codeOffset) 0 codeOffset                           //
+            // 0x80    |  0x80               | DUP          | (codeSize - codeOffset) (codeSize - codeOffset) 0 codeOffset   //
+            // 0x92    |  0x92               | SWAP3        | codeOffset (codeSize - codeOffset) 0 (codeSize - codeOffset)   //
+            // 0x59    |  0x59               | MSIZE        | 0 codeOffset (codeSize - codeOffset) 0 (codeSize - codeOffset) //
+            // 0x39    |  0x39               | CODECOPY     | 0 (codeSize - codeOffset)                                      //
+            // 0xf3    |  0xf3               | RETURN       |                                                                //
+            //---------------------------------------------------------------------------------------------------------------//
+            hex"60_0B_59_81_38_03_80_92_59_39_F3", // returns all code in the contract except for the first 11 (0B in hex) bytes
```

Comment:
here we are building the creation code, which is supposed to return the bytecode of the contract to be created. We want the deployed contract to contain the 0x00 at the start to prevent it from being called.

Reply:
I see I see I see, thank you for explaining here and over call, my misunderstanding here.